apps/gdalalg_raster_calc.cpp

```diff
@@ -602,6 +617,35 @@ bool GDALRasterCalcAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
         return false;
     }
 
+    if (!m_noCheckExpression)
+    {
+        const bool bIsVRT =
+            m_format == "VRT" ||
+            (m_format.empty() &&
+             EQUAL(
+                 CPLGetExtensionSafe(m_outputDataset.GetName().c_str()).c_str(),
+                 "VRT"));
+        const bool bIsGDALG =
+            m_format == "GDALG" ||
+            (m_format.empty() &&
+             cpl::ends_with(m_outputDataset.GetName(), ".gdalg.json"));
+        if (m_format == "stream" || bIsVRT || bIsGDALG)
+        {
+            // Try reading a single pixel to check formulas are valid.
+            std::vector<GByte> dummyData(vrt->GetRasterCount());
```

Comment:
Only other thought I had was modifying the VRT XML to read from a dummy source instead (`/vsidevnull` ? :upside_down_face: ) to avoid a potentially expensive read (poorly chunked dataset, for example)

Reply:
yeah, your right. I had a similar thought. Probably that creating a fake 1x1 GeoTIFF file in /vsimem/ should do the trick

Reply:
> yeah, your right. I had a similar thought. Probably that creating a fake 1x1 GeoTIFF file in /vsimem/ should do the trick

done